src/app.rs

```diff
@@ -189,6 +184,12 @@ impl MatchEvent for App {
             );
 
         for action in actions.iter() {
+            self.store.handle_action(action);
+
+            if let Some(_) = action.downcast_ref::<DownloadFileAction>() {
+                self.notify_downloaded_files(cx);
+            }
+
```

Comment:
These replace the `if Event::Signal = event { ... }` handling that was in this file.